cxx/src/lib/io/fileio.cc

```diff
@@ -83,6 +84,88 @@ long int fwrite_to_file(Seismogram& d, const string dir,const string dfile)
 
 	return(foff);
 }
+std::vector<long int> fwrite_to_file(mspass::seismic::Ensemble<mspass::seismic::TimeSeries>& d, const std::string dir,const std::string dfile)
```

Comment:
This should be a `LoggingEnsemble`.   Not essential here for this writer as inheritance might allow the subclass to work, but I would have to test that to be sure.